### DISCUSSION TITLE
Mixing stm32duino and STM HAL

### DISCUSSION THREAD

**Discussion by ghost** [stm32duino/Arduino_Core_STM32#1359] [Q&A]
Is it possible to mix stm32duino and STM HAL(from system folder) in one project?
Most of libs I use are written for Arduino in C++.

**Comment by fpistm** [comment]
Yes it is possible. Here you can find an example:
https://www.stm32duino.com/viewtopic.php?f=41&t=110

**Reply by ghost** [reply]
I'm not sure how this can help me.

**Reply by fpistm** [reply]
> I'm not sure how this can help me.

The example mixes stm32duino and HAL so don't know what  what more do you expect.

**Reply by ghost** [reply]
Sorry, I wanted to ask can I just put "#include <stm32f_something.h>" into Arduino sketch and then use it's functions? Cannot test it right now. How I see in that topic)your link) I don't see any include calls.

**Reply by fpistm** [reply]
This is already included. Only LL need to be specifically included. Note that not all HAL modules are enabled by default but you can enable them thanks a specific file. https://github.com/stm32duino/wiki/wiki/HAL-configuration#customize-hal-or-variant-definition

**Reply by ghost** [reply]
Thanks!

**Reply by fpistm** [reply]
I forgot to mention that you can also disable Arduino API HAL usage:
https://github.com/stm32duino/wiki/wiki/HAL-configuration#hal-module-only

Som Arduino API already used HAL API which could prevent you to use it (IRQ handler redefinition, msp init,...) so you have to disable it if needed.
